configure.ac

```diff
@@ -2,42 +2,44 @@ AC_INIT(tdnf, 2.0.0)
 AC_MSG_NOTICE([tdnf configuration])
 
 AC_CANONICAL_SYSTEM
-AM_INIT_AUTOMAKE([-Wall -Werror foreign])
+AM_INIT_AUTOMAKE([-Wall foreign])
```

Comment:
It seems the compiler flag changes are independent of the pkgconfig changes. Might I suggest making the compiler flag changes a separate commit or even a separate PR? With separate PRs the pkgconfig and compiler flag discussion and changes could go upstream independently. That's useful because I think removing -Werror *might* be the wrong solution to enable building tndf with other compilers.

I realize they seem trivial but I question that. -Werror, at least with the compiler used when it was added, is likely there for a reason. Did you track it back to the commit that added it? Also: some projects just consider it a part of good code hygiene and turning it off could be short-sighted.

Also, in the commit message for that change, I think it would be nice to have a little more information on the warnings-turned-errors that the code triggers -- talk about the specific warnings that the code triggers and why that justifies turning off -Werror globally rather than:
1) changing the code that triggers the warning
2) turning it off for a specific compiler
3) turning it off for a specific portion of the code that has been reviewed and can't/won't be changed

Reply:
The problem with `-Werror` is that it also triggers warning->error transition for code units that aren't governed by TDNF, and as the compiler changes, it's problematic to keep that in there. This caused issues for me in various environments, so I dropped it to fix that.

IMO, it's a mistake to enforce `-Werror` this way, unless you make it a switch that can be switched off. However, I don't know autotools well enough to make that work.

Reply:
I can split the `-Werror` change into a separate commit if you want, but I don't really know how to make that a switch easily.

Reply:
Looking into this a bit more I realize I potentially attached this comment to the wrong section; from the automake info pages describing these arguments to AM_INIT_AUTOMAKE():
        We are speaking of *Automake* warnings here, such as dubious instructions in ‘Makefile.am’.
        This has absolutely nothing to do with how the compiler will be called

So your change alters both the way automake treats warnings and the way the C compiler treats warnings. In other words, the -Werror here does not propagate into the compiler's CFLAGS.

Again, some output from your attempts to build would be useful here -- if you include that in your commit message I could then tell whether or not the compiler had tripped on -Werror or if it was automake or both. This would also help determine if it's better to change the code to not trigger the warnings rather than remove -Werror. So I strongly encourage you to include some output from the failed builds in your commit message.

Assuming removing the flag is still the way to go there are ways to restrict -Werror to subsets of the build.

You can pull it out of configure.ac and put it in only those Makefile.am that do not trigger errors. That could be alot depending on what you've encountered. Alternately, you can keep -Werror in the top level configure.ac and use -Wno-error in the fewer Makefile.am files which encounter specific warnings-turned-errors.

As to how to add it to Makefile.am, automake offers a few alternatives. You can use it on all the things in that Makefile.am by adjusting AM_CFLAGS. Or if it's one specific executable or library you can adjust the specific foo_CFLAGS variable. See section 27.6 "Flag Variables Ordering" documentation:

https://www.gnu.org/software/automake/manual/html_node/Flag-Variables-Ordering.html#Flag-Variables-Ordering